sdk/src/test/scala/com/horizen/account/api/rpc/service/EthServiceTest.scala

```diff
@@ -30,66 +40,105 @@ class EthServiceTest extends JUnitSuite
+    assertEquals(
+      Numeric.toHexStringWithPrefix(BigInteger.valueOf(params.chainId)),
+      ethService.execute(rpcRequest).asInstanceOf[Quantity].value
+    )
+  }
+
+  @Test
+  def eth_blockNumber(): Unit = {
+    val rpcRequest = buildRpcRequest("eth_blockNumber", null, null)
+    assertEquals("0x1", ethService.execute(rpcRequest).asInstanceOf[Quantity].value)
   }
 
   @Test
-  def testEthService(): Unit = {
+  def invalidJsonRpcData(): Unit = {
     val mapper = new ObjectMapper
 
-    // Test 1: Parameters are of wrong type
-    var json = "{\"id\":\"1\", \"jsonrpc\":\"2.0\",\"method\":\"eth_estimateGas\", \"params\":{\"tx\":\"test\", \"tx2\":\"test2\"}}"
-    var request = mapper.readTree(json)
-    var rpcRequest = new RpcRequest(request)
+    // Test 1: Try to read json request with missing data
+    val json = """{"id":"1", "jsonrpc":"2.0", "params":[]}}"""
+    val request = mapper.readTree(json)
     assertThrows[RpcException] {
-      ethService.execute(rpcRequest)
+      new RpcRequest(request)
     }
 
-    // Test 1: Parameters are of wrong type
-    json = "{\"id\":\"1\", \"jsonrpc\":\"2.0\",\"method\":\"eth_estimateGas\", \"params\":{\"tx\":\"test\", \"tx2\":\"test2\"}}"
-    request = mapper.readTree(json)
-    rpcRequest = new RpcRequest(request)
+    // Test 2: Parameters are of wrong type
+    var rpcRequest = buildRpcRequest("eth_estimateGas", Array("tx", "tx2"), Array("test", "test2"))
     assertThrows[RpcException] {
       ethService.execute(rpcRequest)
     }
 
-    // Test 2: Wrong number of parameters
-    json = "{\"id\":\"1\", \"jsonrpc\":\"2.0\",\"method\":\"eth_estimateGas\", \"params\":[5, 10, 20]}}"
-    request = mapper.readTree(json)
-    rpcRequest = new RpcRequest(request)
+    // Test 3: Wrong number of parameters
+    rpcRequest = buildRpcRequest("eth_estimateGas", null, Array(5, 10, 20))
     assertThrows[RpcException] {
       ethService.execute(rpcRequest)
     }
 
-    // Test 3: Request execution calls correct function and returns value correctly
-    json = "{\"id\":\"1\", \"jsonrpc\":\"2.0\",\"method\":\"eth_chainId\", \"params\":[]}}"
-    request = mapper.readTree(json)
-    rpcRequest = new RpcRequest(request)
-    assertEquals(Numeric.toHexStringWithPrefix(BigInteger.valueOf(params.chainId)), ethService.execute(rpcRequest).asInstanceOf[Quantity].value)
-
-    // Test 4: Trigger IllegalArgumentException rpc call
-    json = "{\"id\":\"1\", \"jsonrpc\":\"2.0\",\"method\":\"eth_estimateGas\", \"params\":[-1]}}"
-    request = mapper.readTree(json)
-    rpcRequest = new RpcRequest(request)
+    // Test 44 Trigger IllegalArgumentException rpc call
+    rpcRequest = buildRpcRequest("eth_estimateGas", null, Array(-1))
     assertThrows[RpcException] {
       ethService.execute(rpcRequest)
     }
   }
 
-  @Test
-  def invalidJsonRpcData(): Unit = {
+  private[this] def buildRpcRequest(method: String, params: Array[String], paramValues: Array[Any]): RpcRequest = {
```

Comment:
I think why not to have too different methods:
1) for named params, that will take argument Map[String, Any], so we will be sure about the key-value relations
2) for raw params, that takes just Array[Any]

It will make test cases a bit more clear.
For example:
`rpcRequest = buildRpcRequest("eth_estimateGas", null, Array(-1))` can be `rpcRequest = buildRpcRequest("eth_estimateGas", Array(-1))`

`var rpcRequest = buildRpcRequest("eth_estimateGas", Array("tx", "tx2"), Array("test", "test2"))`
can be `var rpcRequest = buildRpcRequest("eth_estimateGas", {"tx" -> "test", "tx2" -> "test2"} )`